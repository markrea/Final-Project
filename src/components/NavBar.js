import React from "react";

const NavBar = () => {
    return (
        <div className="navBar">

        </div>
    )
}

export default NavBar;