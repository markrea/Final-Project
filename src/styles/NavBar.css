.navBar{
    display: flex;    
    background-color:rgba(133, 187, 8, 0.582);
    height: inherit;
}
h1{          
    margin-left: 2rem;
    font-family: 'Fredericka the Great', cursive;
    font-weight: lighter;
    font-size: 3em;
    background: -webkit-linear-gradient(rgb(6, 10, 6), rgb(18, 134, 2));
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
}

.navbar-links-item {
    display: inline-block;
    margin: 0;
    padding: 1rem;          
    
}

.menu-item{
    color: white;
    text-decoration: none;
    list-style: none;
    font-size: 1.5rem;
    padding-bottom: 20px; 
}

.bm-burger-button {
    position: fixed;
    width: 36px;
    height: 30px;
    right: 30px;
    top: 20px;
  }
  
  .bm-burger-bars {
    background: #1a2701;
  }
  
  .bm-burger-bars-hover {
    background: #f7f6f8;
  }
  /*
  .bm-cross-button {
    height: 3rem;
    width: 3rem;
    border: 1px solid red;

  }
  
  .bm-cross {
    background: #bdc3c7;

  }*/
  
  .bm-menu-wrap {
    position: fixed;
    height: 100%;

  }
  
 .bm-menu {
    background: #040b2c85;
    padding: 2.5em 1.5em 0;
    font-size: 1.15em;

  }
  
  /* Wrapper for item list */
  .bm-item-list {
    color: #b8b7ad;
    padding: 0.1%;
  }
  
  /* Individual item */
  .bm-item {
    display: inline-block;
  }
  
  /* Styling of overlay */
  .bm-overlay {
    background: rgba(255, 255, 255, 0) !important;
    width: 150% !important;   
    
  }