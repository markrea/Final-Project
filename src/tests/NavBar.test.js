import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  it("component renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });
  it("renders a header", () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByText("Fridge Tracker")).toBeInTheDocument();
  });
  it("renders a navBar div", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByTestId("navBar")).toBeInTheDocument();
  });
  it("displays a logo", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByTestId("logo")).toBeInTheDocument();
  });
  it("displays a list with Home, search by ingredients, recipe, and random recipe", () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByText("Home")).toHaveClass("menu-item");
    expect(getByText("Search by ingredients")).toHaveClass("menu-item");
    expect(getByText("Search by recipe")).toHaveClass("menu-item");
    expect(getByText("Get random recipe")).toHaveClass("menu-item");
  });
  it("contains 4 link elements", () => {
    const { getAllByRole } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    const links = getAllByRole("link");
    expect(links).toHaveLength(4);
  });
  it("should navigate to HomePage with /", () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByText("Home").closest("a")).toHaveAttribute("href", "/");
  });
  it("should navigate to SearchByIngredients page with /search-by-ingredients", () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByText("Search by ingredients").closest("a")).toHaveAttribute(
      "href",
      "/search-by-ingredients"
    );
  });
  it("should navigate to SearchByRecipe page with /search-by-ingredients", () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByText("Search by recipe").closest("a")).toHaveAttribute(
      "href",
      "/search-by-recipe"
    );
  });
  it("should navigate to RandomRecipe page with /get-random-recipe", () => {
    const { getByText } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );
    expect(getByText("Get random recipe").closest("a")).toHaveAttribute(
      "href",
      "/get-random-recipe"
    );
  });
});
