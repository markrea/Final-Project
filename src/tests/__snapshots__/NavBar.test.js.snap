// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders correctly 1`] = `[Function]`;
